### DISCUSSION TITLE
Do not allow selection of ranges containing disabled dates in DatePicker

### DISCUSSION THREAD

**Discussion by dlsso** [mantinedev/mantine#6582] [Q&A]
I would like to prevent users from selecting ranges which contain disabled dates. (I expected this to be the default behavior)

I think the ideal behavior is not allowing selection of anything past a disabled date, the way [react-aria](https://react-spectrum.adobe.com/react-aria/DateRangePicker.html#unavailable-dates) does it, but just throwing an error would work. As-is the rangepicker is not usable for my case.

[Demo of the issue](https://codesandbox.io/p/sandbox/mantine-datepicker-no-types-forked-dlwgr8?workspaceId=eb28aa01-3dd6-45bf-b763-b2f3cff68338&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clz0f2cmg00062v6j9e8ock7z%2522%252C%2522sizes%2522%253A%255B100%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clz0f2cmg00022v6j982dbira%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clz0f2cmg00032v6jeoa3um3l%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clz0f2cmg00052v6jj5b2shst%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clz0f2cmg00022v6j982dbira%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clz0f2cmg00012v6j5sz8ih95%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clz0fnk83000s2v6j4fgwcrkt%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A13%252C%2522startColumn%2522%253A15%252C%2522endLineNumber%2522%253A13%252C%2522endColumn%2522%253A15%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clz0f2cmg00022v6j982dbira%2522%252C%2522activeTabId%2522%253A%2522clz0fnk83000s2v6j4fgwcrkt%2522%257D%252C%2522clz0f2cmg00052v6jj5b2shst%2522%253A%257B%2522id%2522%253A%2522clz0f2cmg00052v6jj5b2shst%2522%252C%2522activeTabId%2522%253A%2522clz0ft99m001j2v6ji4p8qdff%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clz0f2cmg00042v6jyoqyywru%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clz0fq2fe001o2v6jzilv9os2%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clz0ft5dh00192v6jofenpj7r%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522SANDBOX_INFO%2522%252C%2522isCloud%2522%253Afalse%252C%2522id%2522%253A%2522clz0ft99m001j2v6ji4p8qdff%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clz0f2cmg00032v6jeoa3um3l%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clz0f2cmg00032v6jeoa3um3l%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by rtivital** [comment]
You can easily do that on your side: when the first date in range is selected, set `minDate` and `maxDate` to the values of the nearest disabled dates.

**Reply by dlsso** [reply]
Thanks, the suggestion is appreciated. I do think this would still be a valuable feature. It may be easy as in "not tricky" but you still need to capture the event, gather the data from the rangepicker, run a find algo for both values, and set the values, whereas on some date pickers it is the default behavior or a simple flag.
